### DISCUSSION TITLE
Where can I find the documentation/swagger of the console backend APIs?

### DISCUSSION THREAD

**Discussion by alexBonadei** [mia-platform/community#358] [Q&A] [answered]
### Which section?

Console

### Which is the version?

.

### Description

Hi,
I'd like to get informations about the APIs exposed by the backend of the console.

I have this swagger that shows some of them: https://console.cloud.mia-platform.eu/documentations/api-portal/#/api

Where can I find all the endpoints ( `/api/backend/projects/:projectid/branches/` or `/api/backend/projects/:projectid/tags/` for example) ?

Thanks in advance

**Comment by davidebianchi** [comment, marked as answer]
Hi @alexBonadei! The API exposed in the portal are at the moment the publicly exposed API.
We are working to stabilize the API and, once an API is declared stable, we will expose them on the portal. We are working on it to set as public as much API as soon as possible.
If you need some API not present at the moment, you should contact us with your needs and we will give you the needed informations.

**Reply by alexBonadei** [reply]
Ah ok, perfect!

Thank you @davidebianchi!

**Reply by zachhutchins** [reply]
@davidebianchi Is there any update on this? We have several needs for endpoints the console uses that would help with automation. For example, we cannot update configurations via automation i.e. `POST /projects/:projectid/revisions/:revisionId/configuration`.

Is there a timeline for all endpoints to be exposed and stable?

Is there a place where updates to endpoints being exposed will be posted?

**Reply by davidebianchi** [reply]
Hi @zachhutchins! We are still working on it, we will create a timeline of the APIs stabilization as you suggested. Whenever a new endpoint is exposed, we will add it to the Console API portal. Additionally, we can begin including newly exposed APIs in the Release Notes.
We understand that this would facilitate automation, and we will speed up the process!

In the meantime, if you require an API that is not listed in the portal, feel free to contact us for information. We already have some APIs prioritized, but if you have specific needs, we can prioritize those as well.
